bin/build-runtime-library.sh

```diff
@@ -225,14 +215,24 @@ function main () {
       echo "ok - built static library ($arch-$platform): $(basename "$static_library")"
     else
       echo "failed to build $static_library"
+      exit 1
     fi
   else
     if [ -f $static_library ]; then
       echo "ok - using cached static library ($arch-$platform): $(basename "$static_library")"
     else
       echo "static library doesn't exist after cache check passed: ($arch-$platform): $(basename "$static_library")"
+      exit 1
     fi
   fi
+
+  lib_size=$(stat_size $static_library)
+
+  if (( $lib_size < 1048576 )); then
```

Comment:
Why are we checking for the size? Why not just bail when the library fails to build or doesn't exist

Reply:
It's just a fail safe.

I  have seen `llvm-ar` create 8 byte files when `${objects[@]}` was empty.

Sure we could check the array, but a better sanity check is having a decent number of bytes.
We could go further, like dumping the objects from the .a and checking vs the c source, but we're not writing a compiler.
If 1024^2 ends up being too small, the responsible dev can adjust at the time.

I have implemented an exists check in `publish-npm-modules.sh`.

Reply:
this seems way too brittle. why can't we verify a successful build another way?

Reply:
Resolved as discussed.